cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -285,9 +285,9 @@ protected void startUp() throws Exception {
         CConfiguration cConfCopy = cConf;
         contextConfig.set(context, cConfCopy, tx, programJar.toURI(), localizedUserResources);
 
-        LOG.info("Submitting MapReduce Job: {}", context);
         // submits job and returns immediately. Shouldn't need to set context ClassLoader.
         job.submit();
+        LOG.info("Submitted MapReduce Job: {}, with jobId {}", context, job.getJobID());
```

Comment:
Isn't the context (`BasicMapReduceContext`) has the job? Shouldn't we just make it as part of `toString` of the context?